lib/src/utils/dart_symbol_map.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:file/file.dart';
+
+import '../configuration.dart';
+import 'flutter_debug_files.dart';
+
+/// If [configuredPath] is provided, validates and returns the absolute path to the Dart symbol map.
+/// If not provided, returns null.
+///
+/// Note: we do not scan the filesystem for this file because the file does not
+/// have a special extension so worst case we would have to check every file.
+Future<String?> resolveDartSymbolMapPath({
+  required FileSystem fs,
+  String? configuredPath,
+}) async {
+  if (configuredPath == null || configuredPath.isEmpty) return null;
+
+  final file = fs.file(configuredPath);
+  if (!await file.exists()) {
+    throw StateError(
+      "Dart symbol map file not found at '$configuredPath'. Ensure the path is correct and the file exists.",
+    );
+  }
+
+  return file.absolute.path;
+}
+
+/// Finds Flutter-relevant debug file paths for Android and Apple (iOS/macOS)
+/// that should be paired with a Dart symbol map.
+Future<Set<String>> findFlutterRelevantDebugFilePaths({
+  required FileSystem fs,
+  required Configuration config,
+}) async {
+  final Set<String> foundPaths = <String>{};
+
+  Future<void> collectAndroidSymbolsUnder(String rootPath) async {
+    if (rootPath.isEmpty) return;
+
+    final directory = fs.directory(rootPath);
+    if (await directory.exists()) {
+      await for (final entity
+          in directory.list(recursive: true, followLinks: false)) {
+        if (entity is! File) continue;
+        final String basename = fs.path.basename(entity.path);
+        if (basename.startsWith('app') &&
+            basename.endsWith('.symbols') &&
+            !basename.contains('darwin')) {
+          foundPaths.add(fs.file(entity.path).absolute.path);
+        }
+      }
+      return;
+    }
+
+    final file = fs.file(rootPath);
```

Comment:
Is this a case we need? Where the `rootFolder` is a symbol file? It's also duplicating the exact same code from above. Also clashes with naming, as `collectAndroidSymbolsUnder` implies that the param is supposed to be a folder.

Reply:
hm probably not, the user can provide the directory but not really specific symbols, I'll remove it